include/seastar/core/sstring.hh

```diff
@@ -574,9 +574,22 @@ public:
     bool operator!=(const basic_sstring& x) const noexcept {
         return !operator==(x);
     }
+#if __cpp_lib_three_way_comparison
+    constexpr auto operator<=>(const auto& x) const noexcept {
```

Comment:
Please use std::strong_ordering for the return type. It's not good to make readers have to dig for the return type.

Also, conditionally include \<compare>.